srcpkgs/godot/template

```diff
@@ -1,40 +1,41 @@
 # Template file for 'godot'
 pkgname=godot
-version=3.5.1
+version=4.0.3
 revision=1
 archs="x86_64* i686* aarch64* armv7* ppc64*"
 build_style=scons
-# Godot contains private copies of libraries
-# that already have been packaged elsewhere.
-# Use builtin bullet for now as it's too old in repos (needs 2.89)
-# Toggle to not use builtin once bullet has been updated
-make_build_args="platform=x11 tools=yes target=release_debug dev=no progress=no
- builtin_bullet=false builtin_libpng=false builtin_libvpx=false
- builtin_libwebp=false builtin_libogg=false builtin_libtheora=false
- builtin_opus=false builtin_libvorbis=false builtin_enet=false
- builtin_zlib=false builtin_freetype=false builtin_mbedtls=false
- builtin_miniupnpc=false builtin_pcre2=false"
+# Build currently fails with embree-4.X
+make_build_args="platform=linuxbsd target=editor progress=no production=yes
+ builtin_embree=true builtin_enet=false builtin_freetype=false
+ builtin_graphite=false builtin_harfbuzz=false builtin_libogg=false
+ builtin_libpng=false builtin_libtheora=false builtin_libvorbis=false
+ builtin_libwebp=false builtin_mbedtls=false builtin_miniupnpc=false
+ builtin_pcre2=false builtin_zlib=false builtin_zstd=false"
 hostmakedepends="pkg-config clang"
-makedepends="
- alsa-lib-devel freetype-devel glu-devel libXcursor-devel libXi-devel
- libXinerama-devel libXrender-devel libXrandr-devel libX11-devel
- bullet-devel libpng-devel libvpx-devel libwebp-devel libogg-devel libtheora-devel
- opus-devel opusfile-devel libvorbis-devel libenet-devel zlib-devel mbedtls-devel
- miniupnpc-devel pcre2-devel pulseaudio-devel"
+makedepends="alsa-lib-devel freetype-devel mesa glu-devel libXcursor-devel
+ libXi-devel libXinerama-devel libXrender-devel libXrandr-devel libX11-devel
+ libpng-devel libwebp-devel libogg-devel libtheora-devel libvorbis-devel
+ libenet-devel zlib-devel mbedtls-devel miniupnpc-devel pcre2-devel
+ pulseaudio-devel graphite-devel harfbuzz-devel libzstd-devel"
+depends="speech-dispatcher"
 short_desc="Multiplatform 2D and 3D engine"
 maintainer="Nick Hahn <nick.hahn@hotmail.de>"
 license="MIT"
 homepage="https://www.godotengine.org/"
 distfiles="https://github.com/godotengine/godot/archive/${version}-stable.tar.gz"
-checksum=164523c1c8aef0b69b135645794f5bece3f63788556a56aa293c118cde457023
+checksum=8d629aa4910bd0b13354a3b96c253fb62df130ff8acb510c81e23cd100bfdc9f
 nocross=https://build.voidlinux.org/builders/armv7l_builder/builds/6342/steps/shell_3/logs/stdio
 
 CFLAGS+=" -fPIE -fPIC"
 LDFLAGS+=" -pie"
 
+case "$XBPS_TARGET_MACHINE" in
+	*-musl) makedepends+=" libexecinfo-devel speech-dispatcher-devel";;
```

Comment:
Why is speech-dispatcher-devel only needed on musl?

Why is libexecinfo needed when the crash handler is disabled below?

Reply:
That is a good question. I'll try building without `libexecinfo-devel` and add `speech-dispatcher-devel` to the make dependencies.

Reply:
I don't know why the crash handler was disabled in the first place. Older commits don't explain this change.

If it's okay for you I would remove the crash handler change and move speech-dispatcher-devel and libexecinfo-devel to make_depends.

Note: I tested building with musl with the crash handler change still in place. musl does indeed need libexecinfo-devel and glibc doesn't need it for unknown reasons.

Reply:
I've already tested building without the crash handler change successfully for musl and glibc.

Reply:
> glibc doesn't need it for unknown reasons.

Well glibc implements this internally.

Reply:
I've now changed the template and did the following:

1. Get rid of the crash handler change.
2. Only `libexecinfo-devel` will now only be used for musl builds. (Only needed for musl)
3. `speech-dispatcher-devel` is now a dependency for all architectures.

I've successfully build natively x86, x86_64 and x86_64-musl with these changes.